src/btree/bt_sync.c

```diff
@@ -179,22 +179,13 @@ __sync_file(WT_SESSION_IMPL *session, WT_CACHE_OP syncop)
 		 * Set the checkpointing flag to block such actions and wait for
 		 * any problematic eviction or page splits to complete.
 		 */
-		WT_PUBLISH(btree->checkpointing, WT_CKPT_PREPARE);
+		btree->checkpointing = WT_CKPT_PREPARE;
+		WT_FULL_BARRIER();
 
-		/*
-		 * Sync for checkpoint allows splits to happen while the queue
-		 * is being drained, but not reconciliation. We need to do this,
-		 * since draining the queue can take long enough for hot pages
-		 * to grow significantly larger than the configured maximum
-		 * size.
-		 */
-		F_SET(btree, WT_BTREE_ALLOW_SPLITS);
-		ret = __wt_evict_file_exclusive_on(session);
-		F_CLR(btree, WT_BTREE_ALLOW_SPLITS);
-		WT_ERR(ret);
-		__wt_evict_file_exclusive_off(session);
+		(void)__wt_gen_next_drain(session, WT_GEN_EVICT);
```

Comment:
I didn't see an ordering issue that required the `WT_PUBLISH`, what am I missing?

Regardless, I'll put the `WT_PUBLISH` back in, and remove the barrier before `__wt_gen_next_drain`.

Because `checkpointing` is now volatile, writes to it won't be re-ordered earlier (I vaguely recall the reason I put a barrier after the set to `WT_CKPT_RUNNING` was before I made it volatile).